### DISCUSSION TITLE
[@types/node]@18.7.7 causes issues with typescript@4.7.4

### DISCUSSION THREAD

**Discussion by cjones26** [DefinitelyTyped/DefinitelyTyped#61824] [Issues with a @types package] [answered]
Hello,

We discovered today that the latest release of @types/node@18.7.7 that our `tsc` builds are now failing with the following:

```
~/Work/test npm run build

> test@1.0.0 build
> tsc -p tsconfig.build.json

node_modules/@types/node/globals.d.ts:8:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'cause' must be of type 'Error | undefined', but here has type 'unknown'.

8     cause?: unknown;
      ~~~~~

  node_modules/typescript/lib/lib.es2022.error.d.ts:26:5
    26     cause?: Error;
           ~~~~~
    'cause' was also declared here.

node_modules/@types/node/globals.d.ts:19:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'cause' must be of type 'Error | undefined', but here has type 'unknown'.

19     cause?: unknown;
       ~~~~~

  node_modules/typescript/lib/lib.es2022.error.d.ts:22:5
    22     cause?: Error;
           ~~~~~
    'cause' was also declared here.


Found 2 errors in the same file, starting at: node_modules/@types/node/globals.d.ts:8
```

Though this issue only appears when we have `lib: ["ESNext"]` in our `tsconfig.json`.  Removing this line causes the issue to disappear.

Here's an example repo which reproduces the behavior: https://github.com/cjones26/node-types-issue.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment, marked as answer]
hey, we've just added a property definition to Node types, but it seems it's too in flux really to be properly implemented I think. 
It as done here: #61790
the api itself seems to be in flux, breaking stuff (which is why it passed DT tests accross all Node versions, but could fail in the wild):
microsoft/TypeScript#49639
I'll probably revert the change (it's better to not have unstable api, instead of breaking usage)

**Reply by cjones26** [reply]
Gotcha, thanks @peterblazejewicz ... I guess it's just an unfortunate situation :(.

**Reply by peterblazejewicz** [reply]
the change will be merged once CI finishes, thanks!

**Reply by peterblazejewicz** [reply]
merged
